runtime/lua/vim/lsp/_cmd.lua

```diff
@@ -0,0 +1,142 @@
+local lsp = vim.lsp
+
+local M = {}
+
+--- @param filter? vim.lsp.get_clients.Filter
+--- @return string[]
+local function get_client_names(filter)
+  return vim
+    .iter(lsp.get_clients(filter))
+    :map(function(client)
+      return client.name
+    end)
+    :filter(function(name)
+      return vim.lsp.config[name] ~= nil
+    end)
+    :totable()
+end
+
+---@return string[]
+local function get_config_names()
+  local config_names = vim
+    .iter(vim.api.nvim_get_runtime_file('lsp/*.lua', true))
+    ---@param path string
+    :map(function(path)
+      local file_name = path:match('[^/]*.lua$')
+      return file_name:sub(0, #file_name - 4)
+    end)
+    :totable()
+
+  ---@diagnostic disable-next-line: invisible
+  vim.list_extend(config_names, vim.tbl_keys(vim.lsp.config._configs))
+  return vim.list.unique(config_names)
+end
+
+local complete_args = {
+  start = get_config_names,
+  stop = get_client_names,
+  restart = get_client_names,
+}
+
+local function ex_lsp_start(servers)
+  -- Default to enabling all servers matching the filetype of the current buffer.
+  -- This assumes that they've been explicitly configured through `vim.lsp.config`,
+  -- otherwise they won't be present in the private `vim.lsp.config._configs` table.
+  if #servers == 0 then
+    local filetype = vim.bo.filetype
+    ---@diagnostic disable-next-line: invisible
+    for name, _ in pairs(vim.lsp.config._configs) do
```

Comment:
Nit:
```suggestion
    for name in pairs(vim.lsp.config._configs) do
```

Reply:
This one also no longer exists in #36925.